pkg/operator/controller/ingress/status_test.go

```diff
@@ -254,15 +256,18 @@ func TestComputeIngressDegradedCondition(t *testing.T) {
 			},
 			expectIngressDegradedStatus: operatorv1.ConditionTrue,
 			expectRequeue:               true,
+			// Just use the one minute retry duration for this degraded condition
+			expectAfter: time.Minute,
 		},
 		{
 			name: "pods not scheduled for <10m",
 			conditions: []operatorv1.OperatorCondition{
-				cond(
-					IngressControllerPodsScheduledConditionType, operatorv1.ConditionFalse, "", clock.Now().Add(time.Minute*-9)),
+				cond(IngressControllerPodsScheduledConditionType, operatorv1.ConditionFalse, "", clock.Now().Add(time.Minute*-9)),
 			},
 			expectIngressDegradedStatus: operatorv1.ConditionFalse,
 			expectRequeue:               true,
+			// Grace period is 10 minutes, subtract the 9 minute spoofed last transition time
+			expectAfter: time.Minute,
```

Comment:
This is racy because it assumes that 0 time passes between the point in time when this test case is defined with the time value `clock.Now().Add(time.Minute*-9)` and the point in time when `computeIngressDegradedCondition`/`checkConditions` calls `clock.Now()` to compute the remaining grace time for the test case.  

I see you round the time value in the test runner (`if test.expectAfter.Seconds() != e.After().Round(time.Second).Seconds()`), which should reduce the likelihood of hitting the race condition, but I think it still could happen.  

To force the race condition, add `time.Sleep(time.Second)` right before the `for` loop, and then run the tests:
```
go test ./pkg/operator/controller/ingress
--- FAIL: TestComputeIngressDegradedCondition (1.00s)
    status_test.go:450: "pods not scheduled for <10m": expected requeue after 1m0s, got 58.999745214s
    status_test.go:450: "deployment unavailable for <30s": expected requeue after 10s, got 8.999650475s
    status_test.go:450: "deployment minimum replicas unavailable for <60s": expected requeue after 40s, got 38.999604168s
    status_test.go:450: "deployment not all replicas available for <60m": expected requeue after 40m0s, got 39m58.999553896s
    status_test.go:450: "LB provisioning failing <90s": expected requeue after 30s, got 28.999502123s
    status_test.go:450: "DNS failing <30s": expected requeue after 15s, got 13.999467914s
```

To fix the race condition, please make the test use a fake clock by adding the following lines at the start of `TestComputeIngressDegradedCondition`: 
```go
	clock = utilclock.NewFakeClock(time.Time{})
	defer func() {
		clock = utilclock.RealClock{}
	}()
```
This way, we control the passage of time (the hands on the fake clock don't move unless we move them).  Then the raciness is thereby eliminated, and the rounding can be deleted.

Reply:
Done!